pkg/verifier/gcp/entry_point.go

```diff
@@ -23,40 +29,77 @@ const (
 // - find unreachable endpoints & parse output, then terminate instance
 // - return `g.output` which stores the execution results
 func (g *GcpVerifier) ValidateEgress(vei verifier.ValidateEgressInput) *output.Output {
+	// Default to curl.Probe if no Probe specified
+	if vei.Probe == nil {
+		vei.Probe = curl.Probe{}
+		g.Logger.Debug(vei.Ctx, "defaulted to curl probe")
+	}
 	g.Logger.Debug(vei.Ctx, "Using configured timeout of %s for each egress request", vei.Timeout.String())
 	//default gcp machine e2
 	if vei.InstanceType == "" {
 		vei.InstanceType = DEFAULT_INSTANCE_TYPE
 	}
+	if vei.Timeout == 0 {
+		vei.Timeout = DEFAULT_TIMEOUT
+	}
 
+	vei.Timeout = DEFAULT_TIMEOUT
 	// need to set InstanceType here because default is a AWS machine type
 	vei.InstanceType = DEFAULT_INSTANCE_TYPE
 
 	if err := g.validateMachineType(vei.GCP.ProjectID, vei.GCP.Zone, vei.InstanceType); err != nil {
 		return g.Output.AddError(fmt.Errorf("instance type %s is invalid: %s", vei.InstanceType, err))
 	}
 
-	// Fetch the egress URL list as two strings (one for normal URLs, the other
-	// for TLS disabled URLs); note that this is TOTALLY IGNORED by LegacyProbe,
+	// Fetch the egress URL list from github, falling back to local lists in the event of a failure.
+	// Note that this is TOTALLY IGNORED by LegacyProbe,
 	// as that probe only knows how to use the egress URL lists baked into its
 	// AMIs/container images
-	// egressListStr, tlsDisabledEgressListStr, err := egress_lists.GetEgressListAsString(vei.PlatformType, a.AwsClient.Region)
-	// if err != nil {
-	// 	return a.Output.AddError(err)
-	// }
+	egressListYaml := vei.EgressListYaml
+	var egressListStr, tlsDisabledEgressListStr string
+	if egressListYaml == "" {
+		githubEgressList, githubListErr := egress_lists.GetGithubEgressList(vei.PlatformType)
+		if githubListErr == nil {
+			egressListYaml, githubListErr = githubEgressList.GetContent()
+			if githubListErr == nil {
+				g.Logger.Debug(vei.Ctx, "Using egress URL list from %s at SHA %s", githubEgressList.GetURL(), githubEgressList.GetSHA())
+				egressListStr, tlsDisabledEgressListStr, githubListErr = egress_lists.EgressListToString(egressListYaml, map[string]string{})
+			}
+		}
+		var err error
+		if githubListErr != nil {
+			g.Output.AddError(fmt.Errorf("failed to get egress list from GitHub, falling back to local list: %v", githubListErr))
+			egressListYaml, err = egress_lists.GetLocalEgressList(vei.PlatformType)
+			if err != nil {
+				return g.Output.AddError(err)
+			}
+			egressListStr, tlsDisabledEgressListStr, err = egress_lists.EgressListToString(egressListYaml, map[string]string{})
+			if err != nil {
+				return g.Output.AddError(err)
+			}
+		}
+	}
+
+	// Generate the userData file
+	// As expand replaces all ${var} (using empty string for unknown ones), adding the env variables used in userdata.yaml
+	// os.Expand() deletes unknown variables, so we need to add them back in
 	userDataVariables := map[string]string{
```

Comment:
My understanding is that these variables are eventually used to fill in `startup-script.sh`. If my understanding is correct, I think we can clean up a bunch of these variables that aren't used in there.

Reply:
Removed `AWS_REGION` because it was unused. Also added to the GCP startup-script to set proxy environment variables specified by `HTTP_PROXY` `HTTPS_PROXY` and encode specified `CACERT` in /proxy.pem and make executable. Although file is created every time a GCP instance is started, it is only used if `CACERT` is not empty (the curl probe adds it to `CURLOPTS` to be run )